subsys/net/ip/utils.c

```diff
@@ -518,3 +518,230 @@ bool net_header_fits(struct net_pkt *pkt, u8_t *hdr, size_t hdr_size)
 
 	return false;
 }
+
+#if defined(CONFIG_NET_IPV6) || defined(CONFIG_NET_IPV4)
+static bool convert_port(const char *buf, u16_t *port)
+{
+	char *endptr;
+	int tmp;
+
+	tmp = strtol(buf, &endptr, 10);
```

Comment:
`errno` has to be set to `0` before calling `strtol()`.  Might be a good idea to use `strtoul()` instead since it's unsigned anyway.